### PR TITLE
P20-746: Fix the `studio_saySpriteChoices` Blockly function

### PR DESCRIPTION
**Pull request by artem-vavilov**
The function `studio_saySpriteChoices` dynamically invokes the function `Studio.saySprite` with varying arguments. 
The invocation fails when any argument includes an apostrophe (`'`), due to the lack of proper escaping.
| Before  | After |
| ------- | ----- |
| before | after |

- jira ticket: [P20-746](https://codedotorg.atlassian.net/browse/P20-746)

### CODE REVIEW

**Review by wilkie — APPROVED** [review]
This is a good change. The only potential problem is if `this.getFieldValue` returns an integer or something since it will no longer be a double-quoted string. I think the field value is always a string... is that right? If so, or it won't be a problem... LGTM!

File: apps/src/studio/blocks.js [hunk 1]
```diff
@@ -3208,9 +3208,9 @@ exports.install = function (blockly, blockInstallOptions) {
       this.id +
       "', " +
       (this.getFieldValue('SPRITE') || '0') +
-      ", '" +
-      (this.getFieldValue('VALUE') || ' ') +
-      "');\n"
+      ', ' +
+      JSON.stringify(this.getFieldValue('VALUE') || ' ') +
+      ');\n'
```
Comment (mgc1194) [inline apps/src/studio/blocks.js]: I see some of `studio_saySpriteChoices` still uses double quotes (`"`), should we cleanup the rest of the method?
```
"Studio.saySprite('block_id_" +
      this.id +
      "', " +
      (this.getFieldValue('SPRITE') || '0') +
```

Comment (artem-vavilov) [reply apps/src/studio/blocks.js]: I believe it's fine if I understand you correctly.

Screenshot 2024-02-29 at 20 27 45

File: apps/src/studio/blocks.js [hunk 2]
```diff
@@ -3208,9 +3208,9 @@ exports.install = function (blockly, blockInstallOptions) {
       this.id +
       "', " +
       (this.getFieldValue('SPRITE') || '0') +
-      ", '" +
-      (this.getFieldValue('VALUE') || ' ') +
-      "');\n"
```
Comment (daynew) [inline apps/src/studio/blocks.js:3213]: Would this be more readable?
```
  generator.studio_saySpriteChoices = function () {
    const sprite = this.getFieldValue('SPRITE') || '0';
    const value = JSON.stringify(String(this.getFieldValue('VALUE') || ' '));
    // Generate JavaScript for saying (choices version).
    return `Studio.saySprite('block_id_${this.id}', '${sprite}', '${value}');\n`;
  };
```

File: apps/src/studio/blocks.js (hunk 1 above)
Comment (wilkie) [reply apps/src/studio/blocks.js]: I noticed this too and think we should when it makes sense. However, these often have a single quote inside of them which does mean it should probably use a `"` (double quote) to enclose it. The linter _should_ catch any that _should_ be changed.

File: apps/src/studio/blocks.js (hunk 2 above)
Comment (wilkie) [reply apps/src/studio/blocks.js:3213]: The `value` would have the quotes already. The fix would require, perhaps confusingly, but no more than the original change: 

```
  generator.studio_saySpriteChoices = function () {
    const sprite = this.getFieldValue('SPRITE') || '0';
    const value = JSON.stringify(String(this.getFieldValue('VALUE') || ' '));
    // Generate JavaScript for saying (choices version).
    return `Studio.saySprite('block_id_${this.id}', '${sprite}', ${value});\n`;
  };
```

File: apps/src/studio/blocks.js [hunk 3]
```diff
@@ -3202,16 +3202,11 @@ exports.install = function (blockly, blockInstallOptions) {
...
-    return (
-      "Studio.saySprite('block_id_" +
-      this.id +
-      "', " +
-      (this.getFieldValue('SPRITE') || '0') +
-      ', ' +
-      JSON.stringify(String(this.getFieldValue('VALUE') || ' ')) +
-      ');\n'
-    );
+    return `Studio.saySprite('block_id_${this.id}', ${sprite}, ${value});\n`;
```
Comment (wilkie) [inline apps/src/studio/blocks.js:3209]: Interestingly the `sprite` has the opposite behavior as value... looks like the `SPRITE` value was actually allowed to be an integer and not a string. I think the `spriteIndex` in `saySprite` needs to be an integer.

File: apps/src/studio/blocks.js [hunk 4]
```diff
@@ -3202,7 +3202,7 @@ exports.install = function (blockly, blockInstallOptions) {
   };
 
   generator.studio_saySpriteChoices = function () {
-    const sprite = JSON.stringify(String(this.getFieldValue('SPRITE') || '0'));
+    const sprite = this.getFieldValue('SPRITE') || '0';
```
Comment (wilkie) [inline apps/src/studio/blocks.js:3205]: Looks good!

**Comment by wilkie** [comment]
My approval stands. Looks good. I also like the slight refactor. Thanks!

**Pull request merged**